portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -102,6 +106,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/contact.html");
   }
 
+   private String getUserNickname(String id) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Probably can store this on a private class variable rather than getting a new instance each request (Assuming the object/connection doesn't need to be refreshed)

Reply:
I am a little confused on how I would store it as a private class variable. Should I put it in a new java file?

Reply:
I mean that inside your DataServerlet class you add a variable `private DataService datastore` that you reference here instead of recreating it each time.  Though I'm not sure how this object works, and there may be reason to create a new connection for each request. It is fine to leave this as is.

Reply:
Okay!